tools/syz-fillreports/fillreports.go

```diff
@@ -75,7 +75,7 @@ func processReport(dash *dashapi.Dashboard, bugReport *dashapi.BugReport) {
 	// partially processed Report object. This is not the most robust solution, as
 	// it relies on the internals of how the `Symbolize` method works, but it's short
 	// and works now, and furthermore we do not really have other options.
-	rep.Report = bugReport.Report
+	report.SubstituteReport(rep, bugReport.Report)
```

Comment:
Is the guilty file everything we need here?
If yes, then taking into account the size of the comment we have here, and the size of the new comment in report.go and addition of a new public hacky function just for this case, it can make sense to add `report.ExtractGuiltyFile(report []byte) string` directly.

Reply:
You can use this "optional interface" trick to not touch all reporter implementations:
https://github.com/google/syzkaller/blob/master/vm/vm.go#L169-L174

Reply:
Are there any problems with doing this in the dashboard itself? It could just call ExtractGuiltyFile and potentially not even store guilty files.

Reply:
I woudn't say there are any deep problems with that. A small downside is that it adds an extra dependency to the dashboard, though. We also update dashboard much less often than syz-managers. But that's also nothing critical.

The current approach (syz-manager extracts guilty files and passes them alongside the crash info) does not seem to have any clear downsides either. It works absolutely fine and is already in place. It would take time to redo this and remove that info from the DB and it just does not seem reasonable.
 
`syz-fillreports` is supposed to be called only once during feature deployment, I'm not sure it should affect the way we build the rest of the system.

----

Re. `report.ExtractGuiltyFile(report []byte) string`, it could be just fine for `syz-fillreports` purposes as it would behave just the same, but normally we would cut the `reportPrefixLen` bytes from the report and reportPrefixLen` would be non zero for properly extracted reports.

It could be something like `report.ReportToGuiltyFile(report []byte) string` as a parallel means of getting (a good estimate of) this information from only a symbolized report.

Reply:
Let's do report.ReportToGuiltyFile, it looks much cleaner and simpler.